metrics/src/main/java/org/wildfly/extension/metrics/MetricsCollectorService.java

```diff
@@ -49,47 +47,43 @@
     private final Supplier<ModelControllerClientFactory> modelControllerClientFactory;
     private final Supplier<Executor> managementExecutor;
     private final Supplier<ProcessStateNotifier> processStateNotifier;
-    private final List<String> exposedSubsystems;
-    private final String globalPrefix;
+    private Consumer<MetricCollector> metricCollectorConsumer;
 
     private MetricCollector metricCollector;
     private LocalModelControllerClient modelControllerClient;
 
-    static void install(OperationContext context, List<String> exposedSubsystems, String prefix) {
-        ServiceBuilder<?> serviceBuilder = context.getServiceTarget().addService(WILDFLY_COLLECTOR_SERVICE);
+    static void install(OperationContext context) {
+        ServiceBuilder<?> serviceBuilder = context.getServiceTarget().addService(WILDFLY_COLLECTOR);
         Supplier<ModelControllerClientFactory> modelControllerClientFactory = serviceBuilder.requires(context.getCapabilityServiceName(CLIENT_FACTORY_CAPABILITY, ModelControllerClientFactory.class));
         Supplier<Executor> managementExecutor = serviceBuilder.requires(context.getCapabilityServiceName(MANAGEMENT_EXECUTOR, Executor.class));
         Supplier<ProcessStateNotifier> processStateNotifier = serviceBuilder.requires(context.getCapabilityServiceName(PROCESS_STATE_NOTIFIER, ProcessStateNotifier.class));
-        MetricsCollectorService service = new MetricsCollectorService(modelControllerClientFactory, managementExecutor, processStateNotifier, exposedSubsystems, prefix);
+        Consumer<MetricCollector> metricCollectorConsumer = serviceBuilder.provides(WILDFLY_COLLECTOR);
+        MetricsCollectorService service = new MetricsCollectorService(modelControllerClientFactory, managementExecutor, processStateNotifier, metricCollectorConsumer);
         serviceBuilder.setInstance(service)
                 .install();
     }
 
     MetricsCollectorService(Supplier<ModelControllerClientFactory> modelControllerClientFactory, Supplier<Executor> managementExecutor,
-                            Supplier<ProcessStateNotifier> processStateNotifier, List<String> exposedSubsystems, String globalPrefix) {
+                            Supplier<ProcessStateNotifier> processStateNotifier, Consumer<MetricCollector> metricCollectorConsumer) {
         this.modelControllerClientFactory = modelControllerClientFactory;
         this.managementExecutor = managementExecutor;
         this.processStateNotifier = processStateNotifier;
-        this.exposedSubsystems = exposedSubsystems;
-        this.globalPrefix = globalPrefix;
+        this.metricCollectorConsumer = metricCollectorConsumer;
     }
 
     @Override
     public void start(StartContext context) {
-        modelControllerClient = modelControllerClientFactory.get().createClient(managementExecutor.get());
+        modelControllerClient = modelControllerClientFactory.get().createSuperUserClient(managementExecutor.get());
```

Comment:
I don't think we can do this with metrics. :(

Reply:
you are right, I changed the code in https://github.com/wildfly/wildfly/pull/13756/commits/9c2367ec9fad676658cea070bf6e9e8af95270cf to use a regular client and skip any metrics that can not be read due to RBAC